forge/forge/op/embedding.py

```diff
@@ -26,4 +26,4 @@ def Embedding(
         Dictionary of embeddings
     """
 
-    return op("embedding", name, embedding_table, indices).get_tensor()
```

Comment:
I think we have a bug here. You swapped indices, and embedding_table in function deffinition and in this line. Unless you swap input tensors in each call of Embedding function, your two changes cancel eachother. I caught shape mismatch when doing LLama attention that uses embedding op.

Reply:
So everything works fine when you give first operand Indices, and second operand embeding_table, I agree. But when you have some op decomposing to Embedding this won't work. In fact, In LLama3B adv_index decomposes to Embedding, and it is not aware of this swap and creates embedding op with native order of inputs. After that, indices and embeding_table are misinterpreted

Reply:
@dgolubovicTT please help me understand the underlying issues you're facing. 

In sum, this function swaps input order as TTNN expects them in the defined way (otherwise MLIR lower will not work, and results will be invalid). Do you see any issues doing so here? 

If I understood, your concern for this comment is that other decompositions are not updated as well? If that is the case, it's expected. Similary to stack/slice ops, we don't want to update everything in single run as it can become quite a huge change. Therefore, iterative updates and corrections are expected for some types of ops. 

Please let me know if I'm missing some points here.

Reply:
Ok se, this `op("embedding", name, embedding_table, indices)` is not only used here. It is used in decompose of adv_indexing. I caught a bug in LLama where we decompose adv_index to embeding, and pass attributes without swapping. Therefore we get mismatch in shapes... So it is not about iterative change. This broke decompose.

Reply:
Great, thanks for the clarification and good catch! 

@chandrasekaranpradeep can you do a quick follow-up PR to patch this case as it's relevant for Llama 3B model :))

Reply:
Here is the issue to capture the next PR :))

https://github.com/tenstorrent/tt-forge-fe/issues/321

Reply:
Sure @nvukobratTT